x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java

```diff
@@ -272,25 +285,31 @@ public void validate(String tokenPrincipal, SignedJWT signedJWT, ActionListener<
                 );
 
                 jwkSetLoader.reload(ActionListener.wrap(reloadResult -> {
-                    if (false == reloadResult.v1()) {
-                        // No change in JWKSet
-                        logger.debug("Reloaded same PKC JWKs, can't retry verify JWT token [{}]", tokenPrincipal);
+
+                    if (initialJwksVersion != null && Arrays.equals(jwkSetLoader.getContentAndJwksAlgs().sha256(), initialJwksVersion)) {
```

Comment:
this is the actual fix here. instead of looking at if this this thread reloaded the file, we now now look at the shared violatile sha256 value to determine if the file has changed since the last attempt.